### PR TITLE
[subset] Fix sbix subsetting fuzzer bug

### PR DESCRIPTION
**Pull request by ckitagawa-work**
Root cause: reading past end of sbix file.

The issue was due to a misunderstanding I had that the table was fully sanitized. However, it is only partly sanitized, therefore a more rigorous check that offsets are within range of the table was required.

### CODE REVIEW

**Comment by ckitagawa-work** [comment]
The failing check looks to be msan failing due to a timeout, it is likely a flake. I don't have permission to retry though.

File: src/hb-ot-color-sbix-table.hh [hunk 1]
```diff
@@ -133,7 +133,7 @@ struct SBIXStrike
     return hb_blob_create_sub_blob (sbix_blob, glyph_offset, glyph_length);
   }
 
-  bool subset (hb_subset_context_t *c) const
+  bool subset (hb_subset_context_t *c, unsigned int sbix_len, unsigned int strike_offset) const
```
Comment (garretrieger) [inline src/hb-ot-color-sbix-table.hh]: In this method you only ever use the value sbix_len - strike_offset. So I recommend that you just pass one param which is that value. Could be called something like 'available_len'

File: src/hb-ot-color-sbix-table.hh [hunk 2]
```diff
@@ -334,7 +336,8 @@ struct sbix
   bool add_strike (hb_subset_context_t *c,
                    const void *dst_base,
                    LOffsetTo<SBIXStrike>* o,
```
Comment (garretrieger) [inline src/hb-ot-color-sbix-table.hh]: Just notice that the only thing that happens with the 'o' parameter in this call is that it's set to 0. Recommend that you remove it, and do the set to zero in the serialize_strike_offsets method instead.

File: src/hb-ot-color-sbix-table.hh (hunk 1 above)
Comment (ckitagawa-work) [reply src/hb-ot-color-sbix-table.hh]: Done.

File: src/hb-ot-color-sbix-table.hh (hunk 2 above)
Comment (ckitagawa-work) [reply src/hb-ot-color-sbix-table.hh]: Done.

Comment (ckitagawa-work) [reply src/hb-ot-color-sbix-table.hh]: Done.

**Comment by ckitagawa-work** [comment]
PTAL and merge. The failing tests appear to all be timeouts on fuzzers which I think are flakes as they are not the new testcases.

**Comment by garretrieger** [comment]
The fuzzer timeout's a known issue. On my list too look at it. Looks good, will merge.

**Pull request merged**